script.js

```diff
@@ -0,0 +1,621 @@
+    const currentSuit = suits[suitIndex];
+    // make a variable for the current suit's color
+    const currentSuitColor = suitColors[suitIndex];
+    // make a variable for the current suit's symbol
+    const currentSuitSymbol = suitSymbols[suitIndex];
+
+    // loop to create all cards in this suit
+    // rank 1-13
+    for (let rankCounter = 1; rankCounter <= 13; rankCounter += 1) {
+      // Convert rankCounter to string
+      let cardName = `${rankCounter}`;
+      let displayName = `${rankCounter}`;
+
+      // 1, 11, 12 ,13
+      if (cardName === '1') {
+        cardName = 'ace';
+        displayName = 'A';
+      } else if (cardName === '11') {
+        cardName = 'jack';
+        displayName = 'J';
+      } else if (cardName === '12') {
+        cardName = 'queen';
+        displayName = 'Q';
+      } else if (cardName === '13') {
+        cardName = 'king';
+        displayName = 'K';
+      }
+
+      // make a single card object variable
+      const cardData = {
+        name: cardName,
+        suit: currentSuit,
+        rank: rankCounter,
+        suitSymbol: currentSuitSymbol,
+        display: displayName,
+        color: currentSuitColor,
+      };
+
+      // add the card to the deck
+      newDeck.push(cardData);
+    }
+  }
+
+  return newDeck;
+};
+
+// to display output messages informing player about the state of the game
+const displayGameInfo = (info) => {
+  gameInfo.innerHTML = info;
+};
+
+// deal cards to player at start of the game according to handSize
+const dealStartingCards = (cardsData) => {
+  // remove the cards from previous round
+  playerHand = [];
+
+  // deal cards
+  for (let i = 0; i < handSize; i += 1) {
+    playerHand.push(cardsData.pop());
+  }
+};
+
+// make a card element to be appended to the card container and return it
+const makeCardElement = (cardData) => {
+  const suitEl = document.createElement('div');
+
+  suitEl.classList.add(cardData.color, 'suit');
+
+  suitEl.innerText = cardData.suitSymbol;
+
+  const nameEl = document.createElement('div');
+  nameEl.classList.add('name', cardData.color);
+  nameEl.innerText = cardData.display;
+
+  const cardEl = document.createElement('div');
+  cardEl.classList.add('card', 'col-3', 'align-items-center', 'col-md-2');
+
+  cardEl.appendChild(nameEl);
+  cardEl.appendChild(suitEl);
+
+  return cardEl;
+};
+
+// For exchanging cards -------------------------------------------
+// select the card to exchange or unselect it
+const selectOrUnselectCardToExchange = (cardEl, cardToExchange) => {
+  // when player clicks this card and it has not been selected before,
+  // store it in an array of cards that will be exchanged
+  // but if card is selected before,
+  // remove it from the array of cards that will be exchanged.
+  let isCardPresent = false; // false if card has not been selected before
+  if (cardsToExchange.length > 0) { // only check if there are cards in array
+    for (let j = 0; j < cardsToExchange.length; j += 1) {
```

Comment:
Typically `j` is the secondary index, and `i` is the primary index. For non-nested for loops, prefer `i` over `j` by default.